### DISCUSSION TITLE
Missing C-API exports?

### DISCUSSION THREAD

**Discussion by tondrej** [wasmerio/wasmer#2052] [Q&A]
Hi! First of all, kudos for this very interesting project!

I wanted to try embedding wasmer in Pascal/Delphi using the C API.  A basic "hello" example seems to be working so far.

Now I'm going through [WebAssembly C API examples](https://github.com/WebAssembly/wasm-c-api/tree/master/example) and the Wasmer-provided C-API shared library only seems to export a subset of the APIs. Is there a plan to include them all in the future? Thanks!

Here's the list of exports I've found apparently missing:

wasm_config_delete
wasm_externtype_vec_copy
wasm_externtype_vec_delete
wasm_externtype_vec_new
wasm_externtype_vec_new_empty
wasm_externtype_vec_new_uninitialized
wasm_extern_as_func_const
wasm_extern_as_global_const
wasm_extern_as_memory_const
wasm_extern_as_ref
wasm_extern_as_ref_const
wasm_extern_as_table_const
wasm_extern_copy
wasm_extern_get_host_info
wasm_extern_same
wasm_extern_set_host_info
wasm_extern_set_host_info_with_finalizer
wasm_foreign_as_ref
wasm_foreign_as_ref_const
wasm_foreign_copy
wasm_foreign_delete
wasm_foreign_get_host_info
wasm_foreign_new
wasm_foreign_same
wasm_foreign_set_host_info
wasm_foreign_set_host_info_with_finalizer
wasm_func_as_extern_const
wasm_func_as_ref
wasm_func_as_ref_const
wasm_func_copy
wasm_func_get_host_info
wasm_func_same
wasm_func_set_host_info
wasm_func_set_host_info_with_finalizer
wasm_global_as_extern_const
wasm_global_as_ref
wasm_global_as_ref_const
wasm_global_get_host_info
wasm_global_set_host_info
wasm_global_set_host_info_with_finalizer
wasm_instance_as_ref
wasm_instance_as_ref_const
wasm_instance_copy
wasm_instance_get_host_info
wasm_instance_same
wasm_instance_set_host_info
wasm_instance_set_host_info_with_finalizer
wasm_memory_as_extern_const
wasm_memory_as_ref
wasm_memory_as_ref_const
wasm_memory_get_host_info
wasm_memory_set_host_info
wasm_memory_set_host_info_with_finalizer
wasm_module_as_ref
wasm_module_as_ref_const
wasm_module_copy
wasm_module_get_host_info
wasm_module_obtain
wasm_module_same
wasm_module_set_host_info
wasm_module_set_host_info_with_finalizer
wasm_module_share
wasm_name_copy
wasm_name_delete
wasm_name_new
wasm_name_new_empty
wasm_name_new_uninitialized
wasm_ref_as_extern
wasm_ref_as_extern_const
wasm_ref_as_foreign
wasm_ref_as_foreign_const
wasm_ref_as_func
wasm_ref_as_func_const
wasm_ref_as_global
wasm_ref_as_global_const
wasm_ref_as_instance
wasm_ref_as_instance_const
wasm_ref_as_memory
wasm_ref_as_memory_const
wasm_ref_as_module
wasm_ref_as_module_const
wasm_ref_as_table
wasm_ref_as_table_const
wasm_ref_as_trap
wasm_ref_as_trap_const
wasm_ref_copy
wasm_ref_delete
wasm_ref_get_host_info
wasm_ref_same
wasm_ref_set_host_info
wasm_ref_set_host_info_with_finalizer
wasm_shared_module_delete
wasm_table_as_extern_const
wasm_table_as_ref
wasm_table_as_ref_const
wasm_table_get
wasm_table_get_host_info
wasm_table_set
wasm_table_set_host_info
wasm_table_set_host_info_with_finalizer
wasm_table_type
wasm_trap_as_ref
wasm_trap_as_ref_const
wasm_trap_copy
wasm_trap_get_host_info
wasm_trap_same
wasm_trap_set_host_info
wasm_trap_set_host_info_with_finalizer

**Comment by Earthmark** [comment]
I wonder if this is related to #1883 ?
